profiling/src/allocation.rs

```diff
@@ -180,36 +178,34 @@ pub fn allocation_profiling_rinit() {
     ZEND_MM_STATE.with(|cell| {
         let zend_mm_state = cell.get();
 
-        // Safety: `zend_mm_get_heap()` always returns a non null pointer to a valid heap structure
+        // Safety: `zend_mm_get_heap()` always returns a non-null pointer to a valid heap structure
         let heap = unsafe { Some(zend::zend_mm_get_heap()).unwrap() };
 
-        unsafe {
-            (*zend_mm_state).heap = Some(heap);
-        }
+        unsafe { ptr::addr_of_mut!((*zend_mm_state).heap).write(Some(heap)) };
```

Comment:
Just for my understanding: usafe of `ptr::addr_of_mut` removes the creation of an intermediate reference, is that the only reason why you use this?

Reply:
Yes. I can't run this particular thing under MIRI, though, I so I can't be _sure_ it's necessary. Avoiding references to the thread-local means there's less chance we screw up the aliasing rules with our unsafe.